pkg/execution/state/redis_state/queue_test.go

```diff
@@ -1905,6 +1905,129 @@ func TestSharding(t *testing.T) {
 	})
 }
 
+func TestShardLease(t *testing.T) {
+	r := miniredis.RunT(t)
+	rc, err := rueidis.NewClient(rueidis.ClientOption{
+		InitAddress:  []string{r.Addr()},
+		DisableCache: true,
+	})
+	require.NoError(t, err)
+	defer rc.Close()
+	ctx := context.Background()
+
+	sf := func(ctx context.Context, wsID uuid.UUID) *QueueShard {
+		return &QueueShard{
+			Name:               wsID.String(),
+			Priority:           0,
+			GuaranteedCapacity: 1,
+		}
+	}
+	q := NewQueue(rc, WithShardFinder(sf))
+
+	t.Run("Leasing a non-existent shard fails", func(t *testing.T) {
```

Comment:
would this block be flaky since it's run concurrently?
just thinking the lines below could end up running before this, causing the test to fail.

Reply:
We don't mark this as `t.Parallel()` so nothing here runs concurrently.  In our cloud integration tests you're totally right, it'd cause issues.